.devcontainer/devcontainer.json

```diff
@@ -1,44 +1,100 @@
 {
     "name": "Nav2",
-    "build": {
-        "dockerfile": "../Dockerfile",
-        "context": "..",
-        "target": "dever",
-        "cacheFrom": "ghcr.io/ros-navigation/navigation2:main"
-    },
+    "initializeCommand": ".devcontainer/initialize-command.sh dever", // Bakes to tag nav2:devcontainer
+    "image": "nav2:devcontainer",
     "runArgs": [
-        "--name=nav2"
         // "--cap-add=SYS_PTRACE", // enable debugging, e.g. gdb
         // "--ipc=host", // shared memory transport with host, e.g. rviz GUIs
         // "--network=host", // network access to host interfaces, e.g. eth0
         // "--pid=host", // DDS discovery with host, without --network=host
         // "--privileged", // device access to host peripherals, e.g. USB
         // "--security-opt=seccomp=unconfined", // enable debugging, e.g. gdb
+        // "--device=/dev/dri", // enable Intel integrated graphics
+        // "--ulimit", "nofile=1024:4096", // increase file descriptor limit for valgrind
+        // 
+        "--runtime=nvidia", // enable NVIDIA Container Toolkit
```

Comment:
What happens if no NV GPU exists?

Reply:
Then the user should comment out this device option and use the appropriate command for their local hardware, like `--device=/dev/dri` for Intel integrated graphics. Nvidia is just enabled by default as it's so common in robotics and AI development on linux (my own bias). We could leave all hardware acceleration options commented out by default instead, just a minor inconveniences to me.